planner/core/rule_partition_processor.go

```diff
@@ -239,11 +306,17 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 		return ds, nil
 	}
 	// Try to locate partition directly for hash partition.
-	if pi.Type == model.PartitionTypeHash {
+	switch pi.Type {
+	case model.PartitionTypeRange:
+		return s.processRangePartition(ds, pi)
+	case model.PartitionTypeHash:
 		return s.processHashPartition(ds, pi)
+	case model.PartitionTypeList:
+		return s.processListPartition(ds, pi)
+	}
+	if pi.Type == model.PartitionTypeHash {
 	}
 	if pi.Type == model.PartitionTypeRange {
-		return s.processRangePartition(ds, pi)
 	}
```

Comment:
???

Reply:
sorry, it's a mistake, forgot to remove the useless code.